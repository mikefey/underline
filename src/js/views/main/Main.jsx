import React, { Component } from 'react';
import { HashRouter, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducers from 'reducers';
import deviceHelper from 'libs/device/device-helper';
import Resizer from 'global/resizer/ResizerContainer.jsx';
import CollectionContainer from 'views/collection/CollectionContainer.jsx';
import ReaderContainer from 'views/reader/ReaderContainer.jsx';
import TableOfContents from 'views/table-of-contents/TableOfContentsContainer.jsx';
import Info from 'views/info/InfoContainer.jsx';
import Loader from 'ui/loader/LoaderContainer.jsx';
import rootSaga from 'sagas/root-saga';
import indexedDB from 'libs/indexeddb/indexeddb';
import { dbReady } from 'libs/indexeddb/indexeddb-actions.js';
import bookSchema from 'schemas/book';
import style from './Main.scss';

const deviceOS = deviceHelper.getMobileOperatingSystem();
const middlewares = [];
const sagaMiddleware = createSagaMiddleware();

middlewares.push(sagaMiddleware);

const store = createStore(
  reducers,
  applyMiddleware(...middlewares));

sagaMiddleware.run(rootSaga);
indexedDB.init('underline', 1, [bookSchema])
  .then(() => {
    store.dispatch(dbReady());
  })
  .catch((error) => {
    console.log(error);
  });


/**
 * Root component for the app
 */
class Main extends Component {
  /**
   * Renders component
   * @returns {ReactElement} The component
   */
  render() {
    return (
      <Provider store={store}>
        <div className={style.wrapper}>
          <Resizer useScreenDimensions={(deviceOS === 'Android')} />
          <HashRouter>
            <Switch>
              <Route exact path='/' component={CollectionContainer} />
              <Route exact path='/book/:bookId' component={ReaderContainer} />
            </Switch>
          </HashRouter>
          <TableOfContents />
          <Info />
          <Loader />
        </div>
      </Provider>
    );
  }
}

export default Main;
