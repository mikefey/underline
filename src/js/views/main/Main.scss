.wrapper {
  float: left;
  min-height: 100vh;
  position: absolute;
  top: 0;
  width: 100%;
}
