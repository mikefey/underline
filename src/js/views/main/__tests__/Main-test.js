import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import Main from './../Main.jsx';

test('Main component', (t) => {
  const tree = renderToString(
    <Main />,
  );

  t.snapshot(tree);
});
