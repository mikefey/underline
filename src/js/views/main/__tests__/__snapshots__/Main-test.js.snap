// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component 1`] = `"<div class=\\"Main__wrapper___2YZ37xQMwsnUxu5\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1360151273\\"><!-- react-empty: 2 --><div class=\\"Collection__wrapper___Cqx9sa0pwqe_-pP\\" data-reactid=\\"3\\"><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"4\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"5\\"><button class=\\"InfoButton__wrapper___N_oMTzuA2GfeAJV\\" data-reactid=\\"6\\"><div class=\\"InfoButton__icon___1q8MRLZtz3Z3Myx\\" data-reactid=\\"7\\">i</div></button><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"8\\">Your books</h2><div class=\\"AddBooksButton__wrapper___3WnirUQZPEP4sp_\\" data-reactid=\\"9\\"><div data-reactid=\\"10\\"><div class=\\"AddBooksButton__icon___3Vo3Yz3c1Emxo_A\\" data-reactid=\\"11\\"></div></div></div></div></div><div class=\\"Collection__booksWrapper___JImf8enUFiESmgf\\" data-reactid=\\"12\\"><div class=\\"Collection__message___2JDyaCVrQufdN11\\" data-reactid=\\"13\\"><p data-reactid=\\"14\\"><span data-reactid=\\"15\\"><!-- react-text: 16 -->You don&#x27;t have any books in your collection. Click the button above right to add some from your <!-- /react-text --><a href=\\"https://dropbox.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-reactid=\\"17\\">Dropbox</a><!-- react-text: 18 -->.<!-- /react-text --></span></p><p data-reactid=\\"19\\"><!-- react-text: 20 -->Don&#x27;t have any ebooks? Download some for free at <!-- /react-text --><a href=\\"https://www.gutenberg.org/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-reactid=\\"21\\">Project Gutenberg</a><!-- react-text: 22 -->.<!-- /react-text --></p></div></div></div><div class=\\"TableOfContents__wrapper___2FkfbEMJUG4awYB\\" data-reactid=\\"23\\"><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"24\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"25\\"><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"26\\">Table Of Contents</h2><button class=\\"CloseButton__wrapper___2i9TByxUopDp0cZ\\" data-reactid=\\"27\\"><div class=\\"CloseButton__icon___3Schb-DMASl5XVC\\" data-reactid=\\"28\\"><div class=\\"CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"29\\"></div><div class=\\"CloseButton__iconLineTwo___3dpb1YfW7QXAzBv CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"30\\"></div></div></button></div></div><div class=\\"TableOfContents__buttonsWrapper___3n-b4RZYhh_S3yB\\" data-reactid=\\"31\\"></div></div><div class=\\"Info__wrapper___BoibkEk8lqvqZeV\\" data-reactid=\\"32\\"><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"33\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"34\\"><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"35\\">Underline</h2><button class=\\"CloseButton__wrapper___2i9TByxUopDp0cZ\\" data-reactid=\\"36\\"><div class=\\"CloseButton__icon___3Schb-DMASl5XVC\\" data-reactid=\\"37\\"><div class=\\"CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"38\\"></div><div class=\\"CloseButton__iconLineTwo___3dpb1YfW7QXAzBv CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"39\\"></div></div></button></div></div><div class=\\"Info__contentWrapper___23y7LKZxWPqLQh4\\" data-reactid=\\"40\\"><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"41\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"42\\">Contact</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"43\\"><!-- react-text: 44 -->Mike Fey (<!-- /react-text --><a href=\\"mailto:mikefey@protonmail.com\\" data-reactid=\\"45\\">mikefey@protonmail.com</a><!-- react-text: 46 -->)<!-- /react-text --></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"47\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"48\\">Bugs</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"49\\"><!-- react-text: 50 -->Please open an issue or pull request at <!-- /react-text --><a href=\\"https://github.com/mikefey/underline\\" data-reactid=\\"51\\">https://github.com/mikefey/underline</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"52\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"53\\">License</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"54\\"><!-- react-text: 55 -->This software is covered under the <!-- /react-text --><a href=\\"https://www.gnu.org/licenses/gpl-3.0.en.html\\" data-reactid=\\"56\\">GNU General Public License v3.0</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"57\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"58\\">&quot;Add&quot; button icon credit:</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"59\\"><!-- react-text: 60 -->Icon made by <!-- /react-text --><a href=\\"http://picol.org/\\" data-reactid=\\"61\\">Picol</a><!-- react-text: 62 --> from <!-- /react-text --><a href=\\"http://flaticon.com\\" data-reactid=\\"63\\">flaticon.com</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"64\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"65\\">Privacy</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"66\\"><p class=\\"statement\\" data-reactid=\\"67\\">Your privacy is critically important. Your personal information is not stored on our servers.</p><p data-reactid=\\"68\\">It is Underline&#x27;s policy to respect your privacy regarding any information we may collect while operating our app.</p><strong data-reactid=\\"69\\">Website/App Visitors</strong><p data-reactid=\\"70\\">Like most website/app operators, Underline collects non-personally-identifying information of the sort that web browsers and servers typically make available, such as the browser type, language preference, referring site, and the date and time of each visitor request. Underline&#x27;s purpose in collecting non-personally identifying information is to better understand how Underline&#x27;s visitors use the app. From time to time, Underline may release non-personally-identifying information in the aggregate, e.g., by publishing a report on trends in the usage of the app.</p><strong data-reactid=\\"71\\">Gathering of Personally-Identifying Information</strong><p data-reactid=\\"72\\">Certain visitors to Underline website/app choose to interact with Underline in ways that require Underline to gather personally-identifying information. The amount and type of information that Underline gathers depends on the nature of the interaction. Underline collects such information only insofar as is necessary or appropriate to fulfill the purpose of the visitor&#x27;s interaction with Underline. Underline does not disclose personally-identifying information other than as described below. And visitors can always refuse to supply personally-identifying information, with the caveat that it may prevent them from engaging in certain website/app-related activities.</p><strong data-reactid=\\"73\\">Aggregated Statistics</strong><p data-reactid=\\"74\\">Underline may collect statistics about the behavior of visitors to its websites and app. However, Underline does not disclose personally-identifying information other than as described below.</p><strong data-reactid=\\"75\\">IndexedDB</strong><p data-reactid=\\"76\\">IndexedDB is a database that websites and apps use to store information on a visitor&#x27;s computer. The visitor&#x27;s browser provides that data to the website each time the visitor returns. Underline uses IndexedDB to store the app&#x27;s data on the vistor&#x27;s computer  .</p><strong data-reactid=\\"77\\">Privacy Policy Changes</strong><p data-reactid=\\"78\\">Although most changes are likely to be minor, Underline may change its Privacy Policy from time to time, and in Underline&#x27;s sole discretion. Underline encourages visitors to frequently check this page for any changes to its Privacy Policy.</p><p data-reactid=\\"79\\"><!-- react-text: 80 -->Modified from <!-- /react-text --><a href=\\"http://automattic.com/\\" data-reactid=\\"81\\">Automattic</a><!-- react-text: 82 --> Privacy Policy<!-- /react-text --></p></div></div></div></div><div class=\\"Loader__wrapper___3imVhS9YmKq7eOB\\" data-reactid=\\"83\\"><h2 class=\\"Loader__message___37_pDp9j_PMn6_u\\" data-reactid=\\"84\\"></h2><div class=\\"Loader__loader___OksXbA0VgNlfQ7R\\" data-reactid=\\"85\\"></div></div></div>"`;
